jetty-ee10/jetty-ee10-webapp/src/main/java/org/eclipse/jetty/ee10/webapp/MetaInfConfiguration.java

```diff
@@ -767,6 +714,17 @@ protected List<Resource> findExtraClasspathDirs(WebAppContext context)
             .collect(Collectors.toList());
     }
 
+    private Resource ensureJarsAreOpened(WebAppContext context, Resource resource)
+    {
+        if (Resources.isReadable(resource) &&
+            FileID.isJavaArchive(resource.getURI()) &&
+            !"jar".equals(resource.getURI().getScheme()))
+        {
+            return context.getResourceFactory().newJarFileResource(resource.getURI());
```

Comment:
Why do we need to do this? Why can't we just trust that the Resource will be the right type of resource, and that it will open the jar when and if it needs to? Why does the user of the Resource suddenly have to be aware of the internal implementation of Resource and JarFileResource?